Dockerfile

```diff
@@ -0,0 +1,23 @@
+############################
+# STEP 1 build executable binary
+############################
+FROM golang:alpine AS builder
+# Install git.
+# Git is required for fetching the dependencies.
+RUN apk update && apk add --no-cache git
+COPY . $GOPATH/src/insomniacslk/irc-slack
+WORKDIR $GOPATH/src/insomniacslk/irc-slack
+# Fetch dependencies.
+# Using go get.
+RUN go get -d -v
+# Build the binary.
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -installsuffix cgo -ldflags="-w -s" -o /go/bin/irc-slack
```

Comment:
Why forcing it to `amd64`? I am not familiar with Docker but I assume that people may want to build it for other architectures as well (e.g. ARM, to run on a RPi).

Also, `installsuffix` is not required, see https://github.com/golang/go/issues/9344#issuecomment-69944514

Reply:
ah right, i was just copy and pasting from a tutorial, was lazy to type it out all out.

I'll remove it

Reply:
I never remember the CGO stuff offhand, so always copy it from online, i should just start typing it out and remembering it